independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java

```diff
@@ -130,6 +131,9 @@ public BeanGenerator(AnnotationLiteralProcessor annotationLiterals, Predicate<Do
         }
         baseNameBuilder.append("_");
         baseNameBuilder.append(bean.getIdentifier());
+        if (Boolean.TRUE.toString().equals(System.getProperty("quarkus.random-synthetic-bean-names"))) {
```

Comment:
@mkouba @manovotn  wdyt?

Reply:
(this is to allow running tests that need to generate different synthetic beans in one JVM)

Reply:
I think it's a bad idea. We should try to find a different way to fix this.

Reply:
How come you need the property when I see you added `<reuseForks>false</reuseForks>`?
But to be fair, I don't really like this either. +1 for different way but I have to admit I don't have one at hand...

Reply:
the property is only used in MP JWT TCK where `reuseForks = false` doesn't work out of the box because the tests are using testng xml. 
I tried working this around by removing the testng xml and listing the tests in the surefire plugin configuration but then one of the tests starts failing

Reply:
I don't treat this as the ultimate solution but a workaround until we handle reloading synthetic beans properly. But if this fix can wait until that is fixed, I'm okay with waiting.
BTW, does the fact that synthetic beans are not in the application classloader also impact live reload?

Reply:
You can fix it like this for now: https://github.com/quarkusio/quarkus/commit/d85e2d58af1594892bacd84dc3fbd7c5520468c0

Reply:
works like a charm, thank you @stuartwdouglas !
I'll update the PR soon.